geometry/permutation_test.cpp

```diff
@@ -1,11 +1,12 @@
 ﻿
-#include "geometry/permutation.hpp"
-
 #include <vector>
 
 #include "geometry/frame.hpp"
 #include "geometry/identity.hpp"
 #include "geometry/orthogonal_map.hpp"
+// TODO(egg): this should be at the top, but there's a circular dependency due
+// to compatibility test.
```

Comment:
If we actually intended to keep that, perhaps; but forward-declares and include guards to break cycles get really messy, and we want to kill compatibility code soon.